src/pytest_tap/plugin.py

```diff
@@ -161,7 +181,24 @@ def pytest_configure(config: pytest.Config) -> None:
         config.pluginmanager.register(TAPPlugin(config), "tapplugin")
 
 
-def _make_as_diagnostics(report):
+def _make_as_diagnostics(report, tap_logging):
     """Format a report as TAP diagnostic output."""
     lines = report.longreprtext.splitlines(keepends=True)
+
+    if tap_logging in ["log", "all"]:
```

Comment:
Two things:

1. `lines` could be empty, so a guard is needed to make sure the code doesn't raise an `IndexError`.
2. Because `_make_diagnostic` will be called a lot, I think there should be some module level tuples like `SHOW_CAPTURE_LOG`, `SHOW_CAPTURE_STDOUT`, and `SHOW_CAPTURE_ERR`. This will avoid the unnecessary creation of lots of lists and will enable the `showcapture` value check in constant time instead of linear time.

This suggestion below is illustrative and not meant to be committed as is (the comment would be silly to keep):

```suggestion
    if lines and showcapture in SHOW_CAPTURE_LOG:  # where SHOW_CAPTURE_LOG == ("log", "all")
```